apps/browser/qml/pages/components/ToolBar.qml

```diff
@@ -202,8 +202,14 @@ Column {
             id: backIcon
             expandedWidth: toolBarRow.iconWidth
             icon.source: "image://theme/icon-m-back"
-            active: webView.canGoBack && !findInPageActive
-            onTapped: webView.goBack()
+            active: (webView.canGoBack || webView.contentItem.parentId > 0) && !findInPageActive
+            onTapped: {
+                if (webView.canGoBack) {
```

Comment:
The problem with this imo is, that if you hit back a few times (i.e. to get back to your search results), this will then just close your tab and there is no remorse timer to get it back. That would be very annoying. (Also the icon should change to match the action)

Reply:
This back navigation applies only for tabs that got created by JS i.e. parentId is greater than zero. A tab that got created by JS can be closed (window.close()) by the parent JS and well as by the child itself -- any time. Tabs that are root tabs (no parent) cannot be closed with back button over here or from the JS.

The same behaviour have been implemented to Android FF and mobile Safari.

Reply:
We'll consider the icon separately and do some experiments with it.